examples/apps/simple_wallet/workload/simple_wallet_logic.cpp

```diff
@@ -0,0 +1,173 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <string.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <vector>
+
+#include "simple_wallet_logic.h"
+#include "enclave_utils.h"
+
+// Check if given number is valid positive value
+bool SimpleWalletLogic::IsValidAmount(std::string num_str) {
+    int num = 0;
+    try {
+        num = std::stoi(num_str);
+    } catch (...) {
+        return false;
+    }
+    return (num > 0) ? true : false;
+}
+
+// Create account identified by account_name if not created and
+// update wallet balance
+std::string SimpleWalletLogic::DepositMoney(SimpleWalletIoExecutor& sw_io,
+    std::string account_name, std::string amount) {
+    std::string wallet_balance = sw_io.ReadWalletBalance(account_name);
+    if (wallet_balance.empty()) {
+        wallet_balance = sw_io.UpdateWalletBalance(account_name, amount);
```

Comment:
UpdateWalletBalance() returns empty string on error, Please check for error condition in all the places it is called.

Reply:
right, will handle it.